addons/payment_paypal/models/paypal_s2s.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-'8' "-*-"
+
+import base64
+import json
+import logging
+import urllib
+import urllib2
+
+from odoo import api, models, _
+from odoo.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+class AcquirerPaypalS2s(models.Model):
```

Comment:
Is there a specific reason you did not implement `paypal_s2s_form_validate` to check for mandatory field etc.?